### PR TITLE
AllowAnonymous and RefreshSignInAsync

### PR DESCRIPTION
**Pull request by jbagga**
Addresses #735 #736 

Updates Account/Manage files
Adds ShowRecoveryCodes

### CODE REVIEW

**Comment by jbagga** [comment]
@seancpeters Is there a way I can test these with our functional tests in Identity? I have tried manually scaffolding but there seem to be issues with using the local VS Web CodeGeneration package with my changes in it.

**Comment by seancpeters** [comment]
@prafullbhosale - I'm not sure how to answer this, do you have any suggestions?

**Comment by jbagga** [comment]
@seancpeters If you could share how you currently test your Identity scaffolder changes, that may be enough for this PR. Let's touch base on Monday!

**Comment by prafullbhosale** [comment]
Sorry, missed this earlier. When building locally if you just run build script without providing a buildnumber it will always generate the same version (`-t000`). And then even though your changes are built and packaged, nuget won't overwrite the packages in the cache folder. 
Are you hitting problems with this? 

If so, just pass in `/p:buildnumber=xxx` with the build command.

**Comment by jbagga** [comment]
Yes that is what was happening. I have been testing by providing a build number now. 

Separately, is Travis expected to have builds terminated? It seems like it from the history. If we are not concerned about the Travis results, maybe it can be disabled.

**Comment by jbagga** [comment]
Modifying the `.travis.yml` file with `travis_wait` as explained here https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received may help but @ryanbrandenburg may have another suggestion. Although it seems like the build is stalled because no output was received for 10 minutes (total time is longer). We can consider adding some build/test running logging instead.

**Comment by jbagga** [comment]
I have done manual testing of the common scenarios on these changes but have run into the following issues in MVC:
https://github.com/aspnet/Mvc/issues/7665
https://github.com/aspnet/Mvc/issues/7667

Requesting review but will not be merging until the above are resolved.

File: src/VS.Web.CG.Mvc/Templates/Identity/Pages/Account/Manage/Account.Manage.EnableAuthenticator.cs.cshtml [hunk 1]
```diff
@@ -74,15 +80,10 @@ namespace @thisNamespace
             var user = await _userManager.GetUserAsync(User);
             if (user == null)
             {
-                return NotFound($"Unable to load user with ID '{_userManager.GetUserId(User)}'.");
+                NotFound($"Unable to load user with ID '{_userManager.GetUserId(User)}'.");
```
Comment (javiercn) [inline src/VS.Web.CG.Mvc/Templates/Identity/Pages/Account/Manage/Account.Manage.EnableAuthenticator.cs.cshtml]: This seems wrong

Comment (javiercn) [reply src/VS.Web.CG.Mvc/Templates/Identity/Pages/Account/Manage/Account.Manage.EnableAuthenticator.cs.cshtml]: If its on the library, it's probably a bug

**Comment by seancpeters** [comment]
@javiercn - yes, the UI will need to be updated, in the VS codebase.

**Comment by jbagga** [comment]
Yes I have done some basic scenarios testing. Please see my comment above. I have spoken to @seancpeters already about the new razor page. Once this PR is approved I will file an issue to add Account/Manage/ShowRecoveryCodes to the Identity Scaffolder UI for VS.

**Comment by javiercn** [comment]
@seancpeters Would you mind filing a bug on TFS and sharing it on https://github.com/aspnet/external/issues so that we can keep track of it?

**Comment by seancpeters** [comment]
@javiercn @jbagga - I created this issue to track the UI work: https://github.com/aspnet/External/issues/67

File: src/VS.Web.CG.Mvc/Templates/Identity/Pages/Account/Manage/Account.Manage.ChangePassword.cshtml [hunk 2]
```diff
@@ -5,10 +5,10 @@
 }
 
 <h4>@ViewData["Title"]</h4>
-@Html.Partial("_StatusMessage", Model.StatusMessage)
+<partial name="_StatusMessage" model="Model.StatusMessage"/>
```
[the remark this answers is not in the document]
Comment (javiercn) [reply src/VS.Web.CG.Mvc/Templates/Identity/Pages/Account/Manage/Account.Manage.ChangePassword.cshtml]: Don't do random changes on a given PR.
File a bug, bring it up for ask mode, don't try to sneak unrelated changes in a separate PR.

**Pull request merged**